diff --git a/src/lib/platform/EiKeyState.cpp b/src/lib/platform/EiKeyState.cpp
index 87d786ee..d5a4611e 100644
--- a/src/lib/platform/EiKeyState.cpp
+++ b/src/lib/platform/EiKeyState.cpp
@@ -141,6 +141,8 @@ std::uint32_t EiKeyState::convert_mod_mask(std::uint32_t xkb_mask) const
             barrier_mask |= (1 << kKeyModifierBitControl);
         else if (strcmp(XKB_MOD_NAME_ALT, name) == 0)
             barrier_mask |= (1 << kKeyModifierBitAlt);
+        else if (strcmp(XKB_MOD_NAME_LOGO, name) == 0)
+            barrier_mask |= (1 << kKeyModifierBitSuper);
     }
 
     return barrier_mask;
